### PR TITLE
Persist workspace cards/list selection across session

### LINKED ISSUE #625: Make view style persist between page changes
As a user, I click to view all Workspaces as a list. Then I click into a workspace, then back to the list of workspaces, and all of the workspaces are now displayed as Cards. When the user selects a card or list view for a page, it should persist for a while.

### PR DESCRIPTION
**Pull request by MattPutnam**
Fixes #625

### CODE REVIEW

**Comment by zarsky-broad** [comment]
Do we not care about this for other views where there is a mode toggle?

File: src/libs/ajax.js [hunk 1]
```diff
@@ -553,23 +554,27 @@ export const Ajax = controller => {
...
+        ...rest
       })
     }
 
     componentWillUnmount() {
       this.controller.abort()
     }
   }
+
+  return forwardRef((props, ref) => h(AjaxWrapper, { forwardedRef: ref, ...props }))
```
Comment (panentheos) [inline src/libs/ajax.js:579]: Thanks for figuring out how to do this. This looks great!

File: src/pages/workspaces/List.js [hunk 2]
```diff
@@ -129,12 +129,14 @@ const NewWorkspaceCard = pure(({ listView, onClick }) => {
 })
 
 
-export const WorkspaceList = ajaxCaller(class WorkspaceList extends Component {
+const listViewAtom = Utils.atom(false)
```
Comment (panentheos) [inline src/pages/workspaces/List.js]: Good approach. I think we'll probably also want this for other pages that have card views, so we'll need to decide whether to have separate atoms or one aggregate one with keys.

File: src/pages/workspaces/List.js [hunk 3]
```diff
@@ -210,11 +213,11 @@ export const WorkspaceList = ajaxCaller(class WorkspaceList extends Component {
 
   componentDidUpdate() {
     StateHistory.update(_.pick(
-      ['workspaces', 'filter', 'listView'],
+      ['workspaces', 'filter'],
```
Comment (panentheos) [inline src/pages/workspaces/List.js:267]: Does this mean that this setting no longer persists across page refreshes? Are we ok with that?

**Review by panentheos — APPROVED** [review]
Looks good.

**Comment by katevoss** [comment]
I agree with Isaac that the other pages with view mode toggles (Tools and Notebooks) should also persist when switching back and forth.

**Review by elizzhao — APPROVED** [review]

File: src/libs/globals.js [hunk 4]
```diff
@@ -0,0 +1,53 @@
+import _ from 'lodash/fp'
+import { forwardRef } from 'react'
+import { h } from 'react-hyperscript-helpers'
+import * as Utils from 'src/libs/utils'
+import { Component } from 'src/libs/wrapped-components'
+
+
+const globals = Utils.atom(JSON.parse(sessionStorage['globals'] || '{}'))
```
Comment (panentheos) [inline src/libs/globals.js]: I like the approach of using `sessionStorage`, but I don't think it's a good idea to introduce an open-ended 'globals' store at this point. We have some pretty good tooling around atoms that we can leverage for one-off cases like this.

File: src/libs/globals.js [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+    sessionStorage.removeItem('globals')
+  } else {
+    sessionStorage['globals'] = JSON.stringify(v)
+  }
+})
+
+const get = key => globals.get()[key]
+const set = _.curry((key, value) => globals.update(m => ({ ...m, [key]: value })))
+
+export const globalObserver = key => WrappedComponent => {
```
Comment (panentheos) [inline src/libs/globals.js]: This looks an awful lot like `connectAtom`. Perhaps we could export the atom and use that instead?

File: src/pages/workspaces/workspace/Notebooks.js [hunk 6]
```diff
@@ -125,7 +126,7 @@ class NotebookCard extends Component {
   }
 }
 
-const WorkspaceNotebooks = ajaxCaller(wrapWorkspace({
+const WorkspaceNotebooks = ajaxCaller(Globals.globalObserver('notebooksListView')(wrapWorkspace({
```
Comment (panentheos) [inline src/pages/workspaces/workspace/Notebooks.js]: FYI: When the chain of wrappers gets long, we can use `_.flow` to flatten things out.

**Review by panentheos — APPROVED** [review]
This looks really good!

**Pull request merged**